### DISCUSSION TITLE
Accessing a eloquent type object property in Alpine

### DISCUSSION THREAD

**Discussion by robindrost** [livewire/livewire#7297] [1. Help] [answered]
Hello,

I'm trying to access a public property (an eloquent model) from a Livewire component in Alpine, but it returns null in Javascript.

```
class Chat extends Component
{
    public User $user; // This is filled from within blade (parent component)
}
```

In Alpine I use this:

```
document.addEventListener('alpine:init', () => {
    Alpine.data('chat', () => ({
        init() {
            console.log(@this.user)
        },
    }))
})
```

Now the @this.user return null. Is there anyway to access an eloquent object type here or should I always transform it to an array?

**Comment by WilliamPinto-Olmos** [comment]
same thing as if you use `this.$wire.user` instead of `@this.user`? If so, you may try to instantiate the model first, like in mount method or something (maybe the javascript instance of the component 'cast' since it is not instantiated).

```php
public function mount()
{
   $this->user = new User();
}
```

Also it is always recommended working with arrays or DTOs instead of the Model instances since models are complex instances and may lead into lack of performance,

**Reply by robindrost** [reply]
This does not make any difference. Same for creating a new instance in the mount method.
What does work is using the blade @json helper. `const invitee = @json($invitee)`; But this feels kinda hacky.

Eloquent models are supported as properties so I still don't get why they are not available.

**Reply by WilliamPinto-Olmos** [reply]
Yeah that's kinda weird. I know they disabled model binding in V3, so that might be the reason (they may prevent the models from being available on $wire, so they can't be mutated), you can try enabling `legacy_model_binding` on your config file to see if that's the reason.

**Comment by joshhanley** [comment, marked as answer]
@robindrost for security reasons, Livewire doesn't send any Eloquent Model attributes to the front end (imagine if a token or something was accidently exposed...). If you need user data in the front end, I would create a user FormObject or DTO and assign that to a public property instead.

Hope this helps!

**Reply by robindrost** [reply]
@joshhanley Thanks for the clarification! I will use a DTO for now.
